packages/discord.js/src/structures/Message.js

```diff
@@ -30,127 +30,401 @@ const Util = require('../util/Util');
  * @extends {Base}
  */
 class Message extends Base {
-  constructor(client, data) {
+  constructor(client, data = {}) {
     super(client);
+    this.data.channelId = data.channel_id;
+    this.data.guildId = data.guild_id ?? this.channel?.guild?.id ?? null;
+    this._patch(data);
+  }
 
-    /**
-     * The id of the channel the message was sent in
-     * @type {Snowflake}
-     */
-    this.channelId = data.channel_id;
+  /**
+   * The message's id
+   * @type {Snowflake}
+   */
+  get id() {
+    return this.data.id;
+  }
 
-    /**
-     * The id of the guild the message was sent in, if any
-     * @type {?Snowflake}
-     */
-    this.guildId = data.guild_id ?? this.channel?.guild?.id ?? null;
+  /**
+   * The id of the channel the message was sent in
+   * @type {Snowflake}
+   */
+  get channelId() {
+    return this.data.channel_id;
+  }
 
-    this._patch(data);
+  /**
+   * The id of the guild the message was sent in, if any
+   * @type {?Snowflake}
+   */
+  get guildId() {
+    return this.data.guild_id;
   }
 
-  _patch(data) {
-    /**
-     * The message's id
-     * @type {Snowflake}
-     */
-    this.id = data.id;
+  /**
+   * The timestamp the message was sent at
+   * @type {number}
+   */
+  get createdTimestamp() {
+    return DiscordSnowflake.timestampFrom(this.data.id);
+  }
+
+  /**
+   * The type of the message
+   * @type {?MessageType}
+   */
+  get type() {
+    return this.data.type ?? null;
+  }
+
+  /**
+   * Whether or not this message was sent by Discord, not actually a user (e.g. pin notifications)
+   * @type {?boolean}
+   */
+  get system() {
+    if (!this.data.type) return null;
+    return !NonSystemMessageTypes.includes(this.data.type);
+  }
+
+  /**
+   * The content of the message
+   * @type {?string}
+   */
+  get content() {
+    return this.data.content ?? null;
+  }
+
+  /**
+   * The author of the message
+   * @type {?User}
+   */
+  get author() {
+    return this.client.users.resolve(this.data.author.id);
+  }
+
+  /**
+   * Whether or not this message is pinned
+   * @type {?boolean}
+   */
+  get pinned() {
+    if (!this.data.pinned) return null;
+    return Boolean(this.data.pinned);
+  }
```

Comment:
This doesn't look right. Am I missing something?

```suggestion
  get pinned() {
    return this.data.pinned ?? null;
  }
```